test/scripts/jenkins_test_setup.sh

```diff
@@ -0,0 +1,20 @@
+set -e
+
+function post_work() {
+  set +e
+  if [[ -z "$IS_PIPELINE_JOB" ]] ; then
+    node "$KIBANA_DIR/scripts/report_failed_tests"
+  fi
+
+  if [[ -z "$REMOVE_KIBANA_INSTALL_DIR" && -z "$KIBANA_INSTALL_DIR" && -d "$KIBANA_INSTALL_DIR" ]]; then
+    rm -rf "$REMOVE_KIBANA_INSTALL_DIR"
+  fi
+}
+
+trap 'post_work' EXIT
```

Comment:
I always forget, but this line reads as...run the lines after this line, and trap on exit, and run `post_work` as a cleanup step correct?

Reply:
"Run this command when the EXIT signal happens" and then yeah, the rest of the script immediately executes as normal